### ISSUE TITLE
ModuleNotFoundError: No module named 'flit_core'

### ISSUE THREAD

**Issue by JonathanReeve** [nix-community/poetry2nix#606]
### Describe the issue
While trying to open a shell with one dependency (jupyter-book), I get this error: 

```
running build
running build_py

error: builder for '/nix/store/dkkhpf661nvk905ld0s3zs04066qjh3z-python3.9-pyparsing-3.0.8.drv' failed with exit code 2;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'flit_core'
       > 
       > 
       For full logs, run 'nix log /nix/store/dkkhpf661nvk905ld0s3zs04066qjh3z-python3.9-pyparsing-3.0.8.drv'.
error: 1 dependencies of derivation '/nix/store/m1lpws7fwwsnxqcr7d565bvkr22954sw-python3-3.9.12-env.drv' failed to build

```

### Additional context

- `default.nix`/`shell.nix`/`flake.nix`
```
{ pkgs ? import <nixpkgs> {} }:
let
  myAppEnv = pkgs.poetry2nix.mkPoetryEnv {
    projectDir = ./.;
    editablePackageSources = {
      my-app = ./src;
    };
  };
in myAppEnv.env.overrideAttrs (oldAttrs: {
  buildInputs = [ pkgs.hello ];
})

```

- `pyproject.toml`
```
[tool.poetry]
name = "book-computational-literary-analysis"
version = "0.1.0"
description = "Textbook for Introduction to Computational Literary Analysis"
authors = ["Jonathan Reeve <jonathan.reeve@columbia.edu>"]
license = "GPLv3"

[tool.poetry.dependencies]
python = "^3.9"
jupyter-book = "^0.12.3"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"

```

Poetry.lock is too long for GitHub, [so I've uploaded it here as a gist](https://gist.github.com/JonathanReeve/4ff867f64ec3e52e38e5727142c9ef51).

**Comment by NorfairKing** [comment]
I ran into this too, and this overlay got me to a different error:
```
pyparsing = super.pyparsing.overridePythonAttrs (old: {
      buildInputs = (old.buildInputs or [ ]) ++ [
        self.flit-core
      ];
    });
```

**Comment by kenranunderscore** [comment]
Same here with the example from https://www.tweag.io/blog/2020-08-12-poetry2nix/.

**Comment by kenranunderscore** [comment]
As I was told in Matrix: this should be fixed when using the `master` branch of `poetry2nix` (it works for me then).

**Comment by NorfairKing** [comment]
@kenranunderscore any idea which commit fixed it? I can't figure it out by the history.

**Comment by kenranunderscore** [comment]
@NorfairKing Sorry, I have no real idea, but my guess would be https://github.com/nix-community/poetry2nix/commit/b3a52937cc310becf86de214d9092d1d0d119394.

Someone in the NixOS Matrix chat would probably know for sure :)

**Comment by patryk4815** [comment]
I have the same error. I use `flake`, but somehow after updating` nix-channel` it solved the problem - why?

**Comment by cpcloud** [comment]
This was fixed by b3a5293.
